Sources/SwiftDocC/Indexing/Navigator/NavigatorTree.swift

```diff
@@ -270,23 +270,50 @@ public class NavigatorTree {
-     - Parameter interfaceLanguages: A set containing the indication about the interface languages a tree contains.
-     - Parameter atomically: Defines if the read should be atomic.
-     - Parameter presentationIdentifier: Defines if nodes should have a presentation identifier useful in presentation contexts.
+     - Parameters:
+        - url: The file URL from which the tree should be read.
+        - bundleIdentifier: The bundle identifier used to generate the mapping topicID to node on tree.
+        - interfaceLanguages: A set containing the indication about the interface languages a tree contains.
+        - atomically: Defines if the read should be atomic.
+        - presentationIdentifier: Defines if nodes should have a presentation identifier useful in presentation contexts.
+        - postNodeAction: Action to run on a node after it has been read.
      */
-    static func read(from url: URL, bundleIdentifier: String? = nil, interfaceLanguages: Set<InterfaceLanguage>, atomically: Bool = true, presentationIdentifier: String? = nil) throws -> NavigatorTree {
+    static func read(
+        from url: URL,
+        bundleIdentifier: String? = nil,
+        interfaceLanguages: Set<InterfaceLanguage>,
+        atomically: Bool = true,
+        presentationIdentifier: String? = nil,
+        postNodeAction: ((NavigatorTree.Node) -> Void)? = nil
+    ) throws -> NavigatorTree {
         let interfaceLanguageMap = Dictionary(uniqueKeysWithValues: interfaceLanguages.map{ ($0.mask, $0)})
         let path = url.path
         if atomically {
-            return try __readAtomically(from: path, bundleIdentifier: bundleIdentifier, interfaceLanguageMap: interfaceLanguageMap, presentationIdentifier: presentationIdentifier)
+            return try __readAtomically(
+                from: path,
+                bundleIdentifier: bundleIdentifier,
+                interfaceLanguageMap: interfaceLanguageMap,
+                presentationIdentifier: presentationIdentifier,
+                postNodeAction: postNodeAction
+            )
         }
-        return try __read(from: path, bundleIdentifier: bundleIdentifier, interfaceLanguageMap: interfaceLanguageMap, presentationIdentifier: presentationIdentifier)
+        return try __read(
+            from: path,
+            bundleIdentifier: bundleIdentifier,
+            interfaceLanguageMap: interfaceLanguageMap,
+            presentationIdentifier: presentationIdentifier,
+            postNodeAction: postNodeAction
+        )
     }
     
     /// Read a tree by loading the whole data into disk and then process the content.
-    fileprivate static func __readAtomically(from path: String, bundleIdentifier: String? = nil, interfaceLanguageMap: [InterfaceLanguage.ID: InterfaceLanguage], presentationIdentifier: String? = nil) throws -> NavigatorTree {
+    fileprivate static func __readAtomically(
```

Comment:
Not sure why this is double-underscored. Since it's not public we could update it to remove the underscores.